### ISSUE TITLE
Reflection warning, NO_SOURCE_PATH:29:25 - call to method startsWith can't be resolved (target class is unknown).

### ISSUE THREAD

**Issue by dustingetz** [hyperfiddle/hyperfiddle#576]

**Comment by dustingetz** [comment]
```
(def $ (db! "datomic:free://datomic:4334/~david.rolle@gmail.com+1"))
=> #'user/$
(d/q '[:find
       [(pull ?attr [* {:db/valueType [:db/ident],
                        :db/cardinality [:db/ident],
                        :db/unique [:db/ident]}]) ...]
       :where
       [:db.part/db :db.install/attribute ?attr]
       [(hyperfiddle.query/attr-not-archived? $ ?attr)]] 
     $)
NullPointerException   clojure.string/starts-with? (string.clj:364)
```

**Comment by dustingetz** [comment]
Root cause was not addressed, re-opened as #583

**Comment by dustingetz** [comment]
David's domain has a single-segment attribute namespace
![image](https://user-images.githubusercontent.com/124158/45560459-1f129100-b813-11e8-894b-1124fc1d5eec.png)
